### ISSUE TITLE
Heredoc syntax <<- should allow tabbed indention of the heredoc sigil in POSIX shell

### ISSUE THREAD

**Issue by drankinatty** [koalaman/shellcheck#2523]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1009, SC1073, SC1040, SC1072
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

i=1
for label in 'Mod_Name' 'Issue No' 'UT Status' 'Dev ID' 'Loc'; do
    if ! grep -q "^    $((i+=1)). $label:" "$1"; then
        cat <<- EOF
        Proposed message does not satisfy this repository's exacting standards
        It must match the template:
                1. Mod_Name: xxx
                2. Issue No: xxx
                3. UT Status: xxx
                4. Dev ID: xxx
                5. Loc: xxx

        The proposed message does not contain the label: $label" >&2
        EOF
        exit 1
    fi
done

```

#### Here's what shellcheck currently says:
```
Line 6	SC1009: The mentioned syntax error was in this simple command.
Line 6	SC1073: Couldn't parse this here document. Fix to allow more checks.
Line 16	SC1040: When using <<-, you can only indent with tabs.
Line 20	SC1072: Here document was not correctly terminated. Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

Above removing the preceding `tab` before the closing `EOF` satisfies all checks. However, POSIX shell allows a `tab` indention of the heredoc with `<<-` and the leading `tab` (like Makefile, and not `tab` + `spaces`) will be removed.

**Comment by koalaman** [comment]
This works locally, but failed on shellcheck.net because the editor defaulted to soft tabs and translated and copy-pasted tabs into spaces. I've updated it to not use soft tabs so that copy-pasting works as expected. Thanks!
